test/api/test-multithread.c

```diff
@@ -0,0 +1,119 @@
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+
+#include <pthread.h>
+
+#include <hb.h>
+#include <hb-ot.h>
+
+const char *text = "طرح‌نَما";
+const char *path =
+#if defined(__linux__)
+		"/usr/share/fonts/truetype/dejavu/DejaVuSans.ttf";
+#elif defined(_WIN32) || defined(_WIN64)
+		"C:\\Windows\\Fonts\\tahoma.ttf";
+#elif __APPLE__
+		"/Library/Fonts/Tahoma.ttf";
+#endif
+
+int num_iters = 200;
+
+hb_font_t *font;
+
+pthread_mutex_t mutex = PTHREAD_MUTEX_INITIALIZER;
+
+static void *
+thread_func (void *data)
+{
+  hb_buffer_t *buf = (hb_buffer_t *) data;
+
+  pthread_mutex_lock (&mutex);
+  pthread_mutex_unlock (&mutex);
+
+  int i;
+  for (i = 0; i < num_iters; i++)
+  {
+    hb_buffer_clear_contents (buf);
+    hb_buffer_set_direction (buf, HB_DIRECTION_RTL);
+    hb_buffer_set_script (buf, HB_SCRIPT_ARABIC);
+    hb_buffer_add_utf8 (buf, text, sizeof (text), 0, sizeof (text));
+    hb_shape (font, buf, NULL, 0);
+  }
+
+  return 0;
+}
+
+int
+main (int argc, char **argv)
+{
+  int i;
+  int num_threads = 10;
```

Comment:
Do we need it configurable? The code works locally only if is set to 1, showing something apparently.